doc/voq/Reliable_TSA.md

```diff
@@ -0,0 +1,75 @@
+
+# Background
+The TSA feature can be used to shift traffic away from a Sonic Router. This is achieved by applying a Route Policy to BGP which causes it to not advertize any Routes to its Neighbors. This capability is useful when commissioning a new Device  or prior to beginning maintenance operations on a device that may already in production. In a Chassis based system that is made up of multiple Line Cards and a Supervisor Card each running a separate instance of the Sonic software - the TSA feature currently works as follows.
+    1. The TSA/TSB commands on the Line Card updates the CONFIG_DB on the Line Card. This update is picked up and processed by the bgpcfgd process.
+    2. The TSA/TSB commands on the Supervisor causes the Supervisor to SSH to each Line Card in sequence and issue the same command on the Line Card.
+
+This implementation of the commands on the Supervisor is not adequate for a Chassis System. Some issues -
+   1. There could be a decent amount of delay when the First Line Card acts on the command to when the last Line Card acts on it. A Unresponsive or a slow to respond Line Card can cause significant delays to the Line Cards behinnd it.
+   2. An unresponsive Line Card will never receive and act on the command. This will be the case even if the Line is rebooted.
+   3. Using the TSA/TSB commands have the effect of erasing the TSA/TSB configuration on the Line Cards. This makes it difficult to operate different Line Cards in a chassis with different TSA settings.
+
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+This sections describes the requirements for the TSA on the Supervisor and the Line Cards. These requirements only affect the VOQ Chassis systems. Specifically T0/T1 non Chassis system should not be impaced.
+1. The TSA is an independent configuration attribute (tsa_enabled) on the Supervisor host CONFGI_DB and the per asic CONFIG_DB on the Line Card. 
+2. TSA/TSB commands on the Supervisor should only change the Supervisor host config.
+3. TSA/TSB commands on the Line Card should only change the config of the asics on that line card.
+4. Changes to Supervisor config should also be reflected into the CHASSIS_APP_DB. The asic instances of all Line Cards subscribe to the tsa_enabled attribute of the CHASSIS_APP_DB.
+5. The net operational TSA state of BGP on the asics of a LC is a function of the TSA configuration of the Supervisor and the LC
+6. If Supervisor tsa_enabled == TRUE, operational state is TSA(TRUE). In this configuration -
+   A. After the Line card reboots - the operational state will be TSA when the Line card comes back up
+   B. After "Config reload" - the operational state will be TSA.
+   C. After any of the dockers get restarted (crash, service restart etc..) - the operational state will be TSA.
+7. If Supervisor tsa_enabled == FALSE, operational TSA state is controlled by LC tsa_enabled config
```

Comment:
Should we update the TSC command, so that user will know what tsa config is currently enforced, assuming a case when there is tsa configured with true/false on SUP/LC

Reply:
Currently when the TSC command is issued on the Supervisor it issues that command on all the line cards and the combined output is displayed. We probably want to keep this behavior. But maybe we can add an extra line of output indicating whether TSA is configured on the SUP?